File: src/mysqlcookbook/recipes/metadata/get_database_names.rb
#!/usr/bin/ruby
# get_database_names.rb: list names of databases on server

require "Cookbook"

begin
  client = Cookbook.connect
rescue Mysql2::Error => e
  puts "Could not connect to server"
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
end

puts "Get database names from INFORMATION_SCHEMA:"
stmt = "SELECT SCHEMA_NAME
        FROM INFORMATION_SCHEMA.SCHEMATA
        ORDER BY SCHEMA_NAME"
client.query(stmt).each do |row|
    puts row.values[0]
end

client.close

File: src/mysqlcookbook/recipes/metadata/get_database_names.pl
#!/usr/bin/perl
# get_database_names.pl

use strict;
use warnings;
use Cookbook;

# SHOULD IT RETURN AN EMPTY LIST?
# Or should it return a reference/undef?
# The latter course is less ambiguous about whether an error occurs

sub get_database_names
{
my $dbh = shift;
my $ref;

  $ref = $dbh->selectcol_arrayref ("SELECT SCHEMA_NAME
                                    FROM INFORMATION_SCHEMA.SCHEMATA
                                    ORDER BY SCHEMA_NAME");
  return $ref ? @{$ref} : undef;
}

my @names;
my $dbh = Cookbook::connect ();
@names = get_database_names ($dbh);
print "Databases:\n";
print join ("\n", @names), "\n";
$dbh->disconnect ();
